### ISSUE TITLE
Muse

### ISSUE THREAD

**Issue by isamu-isozaki** [huggingface/diffusers#2427]
### Model/Pipeline/Scheduler description

We are working on the training script for [Muse](https://muse-model.github.io/) by google, the progress can be seen [here](https://github.com/lucidrains/muse-maskgit-pytorch/pull/25).
I'm trying to make it in the huggingface format as much as I can so that when it's done, then it should be pretty easy to integrate into diffusers hopefully. 

### Open source status

- [x] The model implementation is available
- [ ] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

https://github.com/lucidrains/muse-maskgit-pytorch

**Comment by patrickvonplaten** [comment]
cc @patil-suraj
